### ISSUE TITLE
Evaluator Compiler the class code everytime？

### ISSUE THREAD

**Issue by hongv** [evrete/evrete#20]
Evaluator Compiler generate and compile the class code everytime？

**Comment by andbi** [comment]
The engine compiles string expressions only once, when a new `Knowledge` instance is created. 

You just opened a new issue, what is it exactly? How to reproduce it? If it's not an issue, then I suggest moving it to the Discussions.

**Comment by hongv** [comment]
i mean in the same JVM process, same string expression can be use the same class.

**Comment by andbi** [comment]
That's exactly what the engine does under the hood whenever it's possible. The default expression parser is just a basic implementation of a SPI (Service Provider Interface), and developers are free to use own implementations of these interfaces.
